cmd/lncli/commands.go

```diff
@@ -2285,14 +2292,32 @@ func sendPayment(ctx *cli.Context) error {
 
 	var rHash []byte
 
-	switch {
-	case ctx.IsSet("payment_hash"):
-		rHash, err = hex.DecodeString(ctx.String("payment_hash"))
-	case args.Present():
-		rHash, err = hex.DecodeString(args.First())
-		args = args.Tail()
-	default:
-		return fmt.Errorf("payment hash argument missing")
+	if ctx.Bool("key_send") {
+		if ctx.IsSet("payment_hash") {
+			return errors.New("cannot set payment hash when using " +
+				"key send")
+		}
+		var preimage lntypes.Preimage
+		if _, err := rand.Read(preimage[:]); err != nil {
+			return err
+		}
+
+		req.DestCustomRecords = map[uint64][]byte{
+			invoices.KeySendRecord: preimage[:],
```

Comment:
Re payload what about also adding a set of optional 32 bytes? This kind of nudges users towards an intended use cases of specifying some sort of account/payment ID which is to be looked up on the backend to accept/deny the payments.

Reply:
People can do that easily through the rpc interface if they want. I am not sure if we should make assumptions like that they require 32 bytes in `lncli`. Also, if we always use the same record id for that, different uses may interfere with each other.